src/pages/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from "react"
+import { Link, useNavigate } from "react-router-dom";
+import FormInput from "../../components/FormInput/FormInput";
+import { addUser } from "../../redux/slices/userSlice";
+import './SignUp.css';
+import { useAppDispatch } from "../../hooks/hooks";
+
+const SignUp = () => {
+  const [values, setValues] = useState({
+    username: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+
+  const dispatch = useAppDispatch();
+  const navigate = useNavigate();
+
+  const inputs = [
+    {
+      id: 1,
+      name: "username",
+      type: "text",
+      placeholder: "Username",
+      errorMessage:
+        "Username should be 3-16 characters and shouldn't include any special character!",
+      label: "Username",
+      pattern: "^[A-Za-z0-9]{3,16}$",
+      required: true,
+    },
+    {
+      id: 2,
+      name: "email",
+      type: "email",
+      placeholder: "Email",
+      errorMessage: "It should be a valid email address!",
+      label: "Email",
+      required: true,
+    },
+    {
+      id: 3,
+      name: "password",
+      type: "password",
+      placeholder: "Password",
+      errorMessage:
+        "Password should be 6-20 characters and include at least 1 letter, 1 number and 1 special character!",
+      label: "Password",
+      pattern: `^(?=.*[0-9])(?=.*[a-zA-Z])(?=.*[!@#$%^&*])[a-zA-Z0-9!@#$%^&*]{6,20}$`,
+      required: true,
+    },
+    {
+      id: 4,
+      name: "confirmPassword",
+      type: "password",
+      placeholder: "Confirm Password",
+      errorMessage: "Passwords don't match!",
+      label: "Confirm Password",
+      pattern: values.password,
+      required: true,
+    },
+  ];
+
+  const handleSubmit = (e: React.SyntheticEvent) => {
+    e.preventDefault();
+    const newUser = {
+      username: values.username,
+      email: values.email,
+      password: values.password,
+      isAuth: false,
+      favorites: [],
+      searchParams: [],
+    };
+
+    dispatch(addUser(newUser));
+    navigate('/signin');
+  };
+
+  const onChange = (e: React.SyntheticEvent) => {
```

Comment:
```suggestion
  const onChange: React.ChangeEventHandler<HTMLInputElement> = (e) => {
```
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/forms_and_events/